Sources/ComplexModule/Documentation.docc/ComplexModule.md

```diff
@@ -0,0 +1,63 @@
+# ``ComplexModule``
+
+Types and operations for working with complex numbers.
+
+## Representation
+
+The `Complex` type is generic over an associated `RealType`; complex numbers
+are represented as two `RealType` values, the real and imaginary parts of the
+number.
+```
+let z = Complex<Double>(1, 2)
+let re = z.real
+let im = z.imaginary
+```
+
+### Memory layout
+
+A `Complex` value is stored as two `RealType` values arranged consecutively
+in memory. Thus it has the same memory layout as:
+- A Fortran complex value built on the corresponding real type (as used
+by BLAS and LAPACK).
+- A C struct with real and imaginary parts and nothing else (as used by
+computational libraries predating C99).
+- A C99 `_Complex` value built on the corresponding real type.
+- A C++ `std::complex` value built on the corresponding real type.
+Functions taking complex arguments in these other languages are not
+automatically converted on import, but you can safely write shims that
+map them into Swift types by converting pointers.
+
+## Real-Complex arithmetic
+
+Because the real numbers are a subset of the complex numbers, many
+languages support arithmetic with mixed real and complex operands.
+For example, C allows the following:
+```c
+#include <complex.h>
+double r = 1;
+double complex z = CMPLX(0, 2); // 2i
+double complex w = r + z;       // 1 + 2i
+```
+The `Complex` type does not provide such mixed operators. There are two
+reasons for this choice. First, Swift generally avoids mixed-type
+arithmetic, period. Second, mixed-type arithmetic operators lead to
+undesirable behavior in common expressions when combined with literal
+type inference. Consider the following example:
+```swift
+let a: Double = 1
+let b = 2*a
```

Comment:
I think it is easier to follow along the arithmetic examples (especially the next one), if this unsupported heterogeneous operator example would re-use the example from C above (and below) – I may be wrong.

```suggestion
import ComplexModule
let r: Double = 1.0
let z = Complex<Double>(0, 2)
let w = r + z // ❌
```

Reply:
The point is to provide an example that would break _if we did have a heterogeneous operator_.

Reply:
True, my example is misleading. Does it make sense to adjust the other two examples, and make them use multiplication over addition?

Reply:
I tried to rearrange stuff to clarify it a little bit, might revisit later.

Reply:
LGTM!